packages/text-input/style.module.css

```diff
@@ -0,0 +1,61 @@
+.root {
+  width: 100%;
+
+  /* theme-light default */
+  --label-text-color: var(--gray-3);
+  --input-text-color: var(--gray-2);
+  --input-border-color: var(--gray-5);
+  --input-background-color: var(--white);
+  --error-text-color: var(--danger);
+
+  &.theme-dark {
+    --label-text-color: var(--gray-3);
+    --input-text-color: var(--gray-5);
+    --input-border-color: var(--gray-2);
+    --input-background-color: var(--gray-1);
+  }
+
+  &.theme-brand {
+    --label-text-color: var(--white);
+    --error-text-color: var(--danger-l1);
+  }
+
+  &.hidden {
+    display: none;
+  }
+}
+
+.label {
+  composes: g-type-label from global;
+  display: block;
+  color: var(--gray-3);
+  margin-bottom: 8px;
+}
+
+.input {
+  font-family: inherit;
+  font-size: 1rem;
+  color: var(--input-text-color);
+  border: 1px solid;
+  border-color: var(--input-border-color);
+  background: var(--input-background-color);
+  border-radius: 2px;
+  padding: 12px;
+  width: 100%;
+  line-height: 1.5em; /* Match .g-type-* line-height for siblings */
+
+  &:placeholder-shown {
+    color: var(--gray-3);
+  }
+
+  &.hasError {
+    border-color: var(--danger);
+  }
+}
+
+.error {
+  composes: g-type-body-small from global;
+  color: var(--danger);
+  line-height: 1.4;
```

Comment:
It seems slightly arbitrary that the error `line-height` is a tiny bit different than the input's (`1.4` for error vs. `1.5em` for the input). 🤔 I wonder if we could align these.

Reply:
Great callout - it does feel a little arbitrary, but I think this might be more of a design concern. Assuming the viewer hasn't scaled their font-size, the `1.4` value also gives us a round `21px` rather than `22.5px`. Quick preview:

<img width="719" alt="CleanShot 2021-09-07 at 17 24 43@2x" src="https://user-images.githubusercontent.com/4624598/132413220-6ff29c14-c10c-4e06-8208-f51b6230faa2.png">

While I definitely don't feel there's a big difference either way, I also don't feel like aligning the numbers gives us a very tangible benefit, and worry that we might be making a (very slight) compromise to design polish.